stdlib/io/src/main/ballerina/src/io/io_error.bal

```diff
@@ -25,4 +25,13 @@ public type ConnectionTimedOutError error<CONNECTION_TIMED_OUT, Detail>;
 public const GENERIC_ERROR = "{ballerina/io}GenericError";
 public type GenericError error<GENERIC_ERROR, Detail>;
 
-public type Error GenericError|ConnectionTimedOutError;
+public const ACCESS_DENIED_ERROR = "{ballerina/io}AccessDeniedError";
+public type AccessDeniedError error<ACCESS_DENIED_ERROR, Detail>;
+
+public const FILE_NOT_FOUND_ERROR = "{ballerina/io}FileNotFoundError";
+public type FileNotFoundError error<FILE_NOT_FOUND_ERROR, Detail>;
+
+public const END_OF_FILE_ERROR = "{ballerina/io}EoF";
+public type EoFError error<END_OF_FILE_ERROR, Detail>;
```

Comment:
According to the convention shouldn't this be `EofError`?

Reply:
When I did a quick search what I saw was, in python, there is `EOFError`[1],
in Java there is `EOFException`[2] and in Rust `UnexpectedEof`.

If the error they intend is the same as our `EoFError`, AFAIU we are good to use our convention.

[1] https://docs.python.org/3/library/exceptions.html
[2] https://docs.oracle.com/javase/7/docs/api/java/io/EOFException.html
[3] https://doc.rust-lang.org/std/io/enum.ErrorKind.html#variant.UnexpectedEof